h1b/src/hil/flash/smart_program.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// A state machine for driving execution of the flash module's smart
```

Comment:
Is this smart programming or just a retry system?

Reply:
"Smart programming" is a feature of the TSMC flash module in H1B, and this driver drives the smart programming functionality.

This code is basically a retry system.